lib/contentTypeParser.js

```diff
@@ -99,20 +99,11 @@ ContentTypeParser.prototype.getParser = function (contentType) {
   const parser = this.cache.get(contentType)
   if (parser !== undefined) return parser
 
-  const parsed = safeParseContentType(contentType)
-
-  // dummyContentType always the same object
-  // we can use === for the comparison and return early
-  if (parsed === defaultContentType) {
-    return this.customParsers.get('')
-  }
-
   // eslint-disable-next-line no-var
   for (var i = 0; i !== this.parserList.length; ++i) {
     const parserListItem = this.parserList[i]
-    if (compareContentType(parsed, parserListItem)) {
-      const parser = this.customParsers.get(parserListItem.name)
-      // we set request content-type in cache to reduce parsing of MIME type
+    if (contentType.indexOf(parserListItem) === 0) {
```

Comment:
Don't do it like what I'm doing here, but just to ask - is this value trimmed and lowercased somewhere else?

Maybe we should do that before checking to increase matches? It is still valid to have spaces around the contentType value, and definitely fine to make it case insensitive AFAIK

```suggestion
    if (contentType.trim().toLowerCase().indexOf(parserListItem) === 0) {
```

Reply:
I would go for one purpose at a time.

Reply:
@mcollina were you talking about this part, that runs at every incoming request, when you said node already lowercases the contentType value?

But I guess it doesn’t trim it right?

Would trimming add that much overhead? We should bench but I’m guessing this PR + trimming have less overhead than main right now

Reply:
Generally speaking, `headers` always in lowercase when it comes.
`.trim` only cause problem when it first arrive, if we cache the incoming header.